packages/strapi-plugin-i18n/admin/src/containers/SettingsPage/LocaleSettingsPage.js

```diff
@@ -47,9 +48,11 @@ const LocaleSettingsPage = ({
         <LocaleList
           canUpdateLocale={canUpdateLocale}
           canDeleteLocale={canDeleteLocale}
-          canCreateLocale={canCreateLocale}
+          onToggleCreateModal={canCreateLocale ? setIsOpenedCreateModal : undefined}
         />
       ) : null}
+
+      <ModalCreate isOpened={isOpenedCreateModal} onClose={() => setIsOpenedCreateModal(false)} />
```

Comment:
I guess here you'll need an `handleClose` handler because you don't need to set the `isOpenedCreateModal` state to `false` since the onClose function will be called once the modal is already closed.